src/core/net/srp_server.hpp

```diff
@@ -492,6 +509,21 @@ class Server : public InstanceLocator, private NonCopyable
          */
         void GetLeaseInfo(LeaseInfo &aLeaseInfo) const;
 
+        /**
+         * This method processes the TTL value received in a resource record.
+         *
+         * If no TTL value is stored, this method wil set the stored value to @p aTtl and return `kErrorNone`.
+         * If a TTL value is stored and @p aTtl equals the stored value, this method returns `kErrorNone`.
+         * Otherwise, this method returns `kErrorRejected`.
+         *
+         * @param[in]  aTtl  The TTL received in a resource record.
+         *
+         * @retval kErrorNone      If no TTL value is stored or @p aTtl equals the stored TTL value.
+         * @retval kErrorRejected  If @p aTtl does not equal the stored TTL.
+         *
+         */
+        Error ProcessTtl(uint32_t aTtl);
```

Comment:
Can we declare this method as `private` on `Host`? 

I think it will be called only from `Srp::Server` (and its nested types) and `Server` is already is a `friend` of `Host` so it should work already.

(we can keep the documentation and add it in cpp file?).